### PR TITLE
Add version 3.1.10 for imath library

### PR DESCRIPTION
**Pull request by irieger**
Specify library name and version:  **imath/3.1.10**

Just bumping the version of imath to the newest release

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/imath/all/conanfile.py [hunk 1]
```diff
@@ -45,6 +47,17 @@ def validate(self):
...
+        # Check for unsupported gcc 7 with version 3.1.10
+        compiler = self.settings.compiler
+        compiler_name = str(compiler)
+        is_invalid_gcc = (compiler_name == 'gcc' and Version(compiler.version) < '7')
+        is_version_3_1_10 = (Version(self.version) == '3.1.10')
+        if is_invalid_gcc and is_version_3_1_10:
+            raise ConanInvalidConfiguration(
+                f"Minimum required gcc version: 7 to include infinite in <cmath> with imath 3.1.10."
+            )
+
```
Comment (RubenRBS) [inline recipes/imath/all/conanfile.py:59]: What about backporting https://github.com/AcademySoftwareFoundation/Imath/pull/371 and https://github.com/AcademySoftwareFoundation/Imath/pull/368 to enable old gcc behaviour? I know that supporting such an old gcc release is quite tedious, but turns out it's still a popular version :$

Comment (RubenRBS) [reply recipes/imath/all/conanfile.py:59]: I can do the patching for you if needed :)

**Review by RubenRBS — COMMENTED** [review]
Thanks! Question about the validation vs backporting

File: recipes/imath/all/conanfile.py (hunk 1 above)
Comment (irieger) [reply recipes/imath/all/conanfile.py]: Thanks for offering. I just added a patch derived from the two changes.

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**